test/unittests/state_transition_eof_create_test.cpp

```diff
@@ -925,6 +914,327 @@ TEST_F(state_transition, eofcreate_call_created_contract)
     expect.post[create_address].nonce = 1;
 }
 
+TEST_F(state_transition, creation_tx)
+{
+    rev = EVMC_PRAGUE;
+    const auto deploy_container = eof_bytecode(bytecode(OP_INVALID));
+
+    const auto init_code = returncontract(0, 0, 0);
+    const bytecode init_container = eof_bytecode(init_code, 2).container(deploy_container);
+
+    tx.data = init_container;
+
+    expect.post[Sender].nonce = pre.get(Sender).nonce + 1;
+    const auto create_address = compute_create_address(Sender, pre.get(Sender).nonce);
+    expect.post[create_address].code = deploy_container;
+    expect.post[create_address].nonce = 1;
+}
+
+TEST_F(state_transition, creation_tx_deploy_data)
+{
+    rev = EVMC_PRAGUE;
+    const auto deploy_data = "abcdef"_hex;
+    const auto deploy_container = eof_bytecode(bytecode(OP_INVALID)).data(deploy_data);
+
+    const auto init_code = returncontract(0, 0, 0);
+    const bytecode init_container = eof_bytecode(init_code, 2).container(deploy_container);
+
+    tx.data = init_container;
+
+    expect.post[Sender].nonce = pre.get(Sender).nonce + 1;
+    const auto create_address = compute_create_address(Sender, pre.get(Sender).nonce);
+    expect.post[create_address].code = deploy_container;
+    expect.post[create_address].nonce = 1;
+}
+
+TEST_F(state_transition, creation_tx_auxdata_in_calldata)
+{
+    rev = EVMC_PRAGUE;
+    const auto deploy_data = "abcdef"_hex;
+    const auto aux_data = "aabbccddeeff"_hex;
+    const auto deploy_data_size = static_cast<uint16_t>(deploy_data.size());
+    const auto deploy_container =
+        eof_bytecode(bytecode(OP_INVALID)).data(deploy_data, deploy_data_size);
```

Comment:
yes, but the `aux_data` name is confusing, I only now realized this, as I started figuring out. Should be `input` or something. This is the creation-tx-appended data, so not part of the data section

Reply:
ah, correction - it is the data which becomes aux_data in RETURNCONTRACT, as such, it could have been included in the size. I think I'll add a new test to cover this as well. The difference is in one the aux data is static, in the other it is dynamic.